GraphWebApi/Controllers/SnippetsController.cs

```diff
@@ -74,7 +74,7 @@ public async Task<IActionResult> PostAsync(string lang = "c#", string generation
         {
             // Default to openapi generation if supported and not explicitly requested for.
             if (string.IsNullOrEmpty(generation))
-                generation = OpenApiSnippetsGenerator.SupportedLanguages.Contains(lang) ? "openapi" : "odata";
+                generation = (lang != "java" && OpenApiSnippetsGenerator.SupportedLanguages.Contains(lang)) ? "openapi" : "odata";
```

Comment:
```suggestion
                generation = (!"java" .Equals(lang,StringComparison.OrdinalIgnoreCase)&& OpenApiSnippetsGenerator.SupportedLanguages.Contains(lang)) ? "openapi" : "odata";
```
The dictionary is case insensitive as well so we need to guard against that https://github.com/microsoftgraph/microsoft-graph-devx-api/blob/3c66e2491cc5c32dce561ab24c786b4d06223847/CodeSnippetsReflection.OpenAPI/OpenAPISnippetsGenerator.cs#L77
Same for the other entry point as well.